src/main/java/org/intellij/sdk/codesync/files/LockFile.java

```diff
@@ -0,0 +1,173 @@
+package org.intellij.sdk.codesync.files;
+
+import org.intellij.sdk.codesync.Constants;
+import org.intellij.sdk.codesync.exceptions.FileNotCreatedError;
+import org.intellij.sdk.codesync.exceptions.InvalidYmlFileError;
+import org.intellij.sdk.codesync.utils.CommonUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+A file that will contain information about different kinds of inter-process and inter-thread synchronization data.
```

Comment:
👍 for the detailed comment. 

Since we have 2 types of locks initially
1- can_populate_buffer (Across different IDEs)
2- can_send_diffs (Across multiple instances of same IDE)
(I am not sure why do we need auth lock?)
Maybe we can start off with these 2?
```
   populate_buffer:
        expiry: <time-stamp-indicating-lock-expiry>
        identifier: <name-of-the-project-holding-the-lock>
   send_diffs:
       intellij:
            expiry: <time-stamp-indicating-lock-expiry>
            identifier: <name-of-the-project-holding-the-lock>
       vscode:
            expiry: <time-stamp-indicating-lock-expiry>
            identifier: <name-of-the-project-holding-the-lock>       
```
e.g. What I have normally is 
- `codesyncserver` is opened in PyCharm 
- `codesync-vs` is opened in VSCode
PyCharm will acquire both `populate_buffer` lock and `send_diffs` lock. 
When VSCode is opened, it should only acquire `send_diffs` lock so that other Windows of VSCode do not send multiple diffs. 

The idea of yml file is good, we can use this for Project Specfic Lock as well, e.g. If same repo is opnened in WebStorm and VSCode (Different IDEs), only 1 of them should listen to real-time events otherwise I think data will get messy.